lib/tests/maintainers.nix

```
@@ -0,0 +1,97 @@
+# to run these tests:
+# nix-build nixpkgs/lib/tests/maintainers.nix
+# If nothing is output, all tests passed
+{ pkgs ? import ../.. {} }:
+
+let
+  inherit (pkgs) lib;
+  inherit (lib) types;
+
+  maintainerModule = { config, ... }: {
+    options = {
+      name = lib.mkOption {
+        description = "Your, preferably real, name.";
+        type = types.str;
+      };
+      email = lib.mkOption {
+        description = "Email address.";
+        type = types.str;
+      };
+      github = lib.mkOption {
+        description = ''
+          GitHub handle, as it appears in the URL of your profile page,
+          `https://github.com/<userhandle>`
+        '';
+        type = types.nullOr types.str;
+        default = null;
+      };
+      githubId = lib.mkOption {
+        description = ''
+          GitHub user ID, which can be found at
+          `https://api.github.com/users/<userhandle>`. If `github` is specified
+          but `githubId` isn't, an error is thrown with the correct user ID.
+        '';
+        type = types.nullOr types.ints.unsigned;
+        default = null;
+      };
+      keys = lib.mkOption {
+        description = ''
+          PGP/GPG keys. Add only if you actually use them to sign commits
+          and/or mail.
+
+          To get the required PGP/GPG values for a key run
+          ```shell
+          gpg --keyid-format 0xlong --fingerprint <email> | head -n 2
+          ```
+
+          !!! Note that PGP/GPG values stored here are for informational purposes only, don't use this file as a source of truth.
+        '';
+        type = types.listOf (types.submodule {
+          options.longkeyid = lib.mkOption { type = types.str; };
+          options.fingerprint = lib.mkOption { type = types.str; };
+        });
+        default = [];
+      };
+    };
+  };
+
+  checkMaintainer = handle: uncheckedAttrs:
+  let
+      prefix = [ "lib" "maintainers" handle ];
+      checkedAttrs = (lib.modules.evalModules {
+        inherit prefix;
+        modules = [
+          maintainerModule
+          {
+            _file = toString ../../maintainers/maintainer-list.nix;
+            config = uncheckedAttrs;
+          }
+        ];
+      }).config;
+
+      checkGithubId = lib.optional (checkedAttrs.github != null && checkedAttrs.githubId == null) ''
+        echo ${lib.escapeShellArg (lib.showOption prefix)}': If `github` is specified, `githubId` must be too.'
+        # Calling this too often would hit non-authenticated API limits, but this
+        # shouldn't happen since such errors will get fixed rather quickly
+        info=$(curl -sS https://api.github.com/users/${checkedAttrs.github})
+        id=$(jq -r '.id' <<< "$info")
```

Since the `info` variable is unused I guess one could assign directly to `id` using some piping:
```suggestion
        id=$(curl -sS https://api.github.com/users/${checkedAttrs.github} | jq -r '.id')
```

---

I've had to debug this a bit, which I did with `echo "$info"` inbetween those lines. I think this might come in handy in the future, and it keeps the line width down too.

---

Fair enough. I think it overall looks very nice. Good job!